ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/AbstractRequestHandler.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.requesthandler;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.GATEWAY_TIMEOUT;
+import static io.netty.handler.codec.http.HttpResponseStatus.INTERNAL_SERVER_ERROR;
+
+import java.net.ConnectException;
+import java.time.Duration;
+import java.util.concurrent.TimeoutException;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public abstract class AbstractRequestHandler {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final Duration httpRequestTimeout;
+
+  public AbstractRequestHandler(Duration httpRequestTimeout) {
```

Comment:
Think composition would work better i.e. move these until a request utils class. The transmitter isn't really a request handler and also the sendTransactionHandler doesn't use this but from the name would expect it to do so.

Reply:
Agreed - changed this to a VertxRequestTransmitter - means all the logs etc. all just fit ...